### PR TITLE
Switch to using RCUTILS_LOG_ERROR for depthimage_to_laserscan.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

File: src/DepthImageToLaserScanROS.cpp [hunk 1]
```diff
@@ -92,7 +91,7 @@ void DepthImageToLaserScanROS::infoCb(sensor_msgs::msg::CameraInfo::SharedPtr in
 void DepthImageToLaserScanROS::depthCb(const sensor_msgs::msg::Image::SharedPtr image)
 {
   if (nullptr == cam_info_) {
-    ROS_ERROR("No camera info, skipping point cloud squash\n");
+    RCUTILS_LOG_ERROR("No camera info, skipping point cloud squash\n");
```
Comment (dhood) [inline src/DepthImageToLaserScanROS.cpp]: you probably don't want to add the newline manually anymore (and below)

Comment (clalancette) [reply src/DepthImageToLaserScanROS.cpp]: Good call, I'll fix that.

File: src/DepthImageToLaserScanROS.cpp [hunk 2]
```diff
@@ -92,7 +91,7 @@ void DepthImageToLaserScanROS::infoCb(sensor_msgs::msg::CameraInfo::SharedPtr in
 void DepthImageToLaserScanROS::depthCb(const sensor_msgs::msg::Image::SharedPtr image)
 {
   if (nullptr == cam_info_) {
-    ROS_ERROR("No camera info, skipping point cloud squash\n");
+    RCUTILS_LOG_ERROR("No camera info, skipping point cloud squash");
```
Comment (mikaelarguedas) [inline src/DepthImageToLaserScanROS.cpp]: please also remove the trailing semi-colon in the macro calls

File: src/DepthImageToLaserScanROS.cpp [hunk 3]
```diff
@@ -103,6 +102,6 @@ void DepthImageToLaserScanROS::depthCb(const sensor_msgs::msg::Image::SharedPtr
   }
   catch (std::runtime_error& e)
   {
-    ROS_ERROR("Could not convert depth image to laserscan: %s\n", e.what());
+    RCUTILS_LOG_ERROR("Could not convert depth image to laserscan: %s", e.what());
```
Comment (dhood) [inline src/DepthImageToLaserScanROS.cpp]: feel free to put this back to throttled as it is in ROS1 as this is available in RCUTILS

**Comment by mikaelarguedas** [comment]
We avoided changing the macro names in the ported code to keep the diff minimal with upstream ROS 1 package. Would it make sense to just change the define rather than changing the invocation?

File: src/DepthImageToLaserScanROS.cpp (hunk 3 above)
Comment (clalancette) [reply src/DepthImageToLaserScanROS.cpp]: Will do.

**Comment by clalancette** [comment]
> We avoided changing the macro names in the ported code to keep the diff minimal with upstream ROS 1 package. Would it make sense to just change the define rather than changing the invocation?

Yeah, that is probably a better idea on a second think.  I'll switch back to defining the ROS_* macros in terms of the RCUTILS macros.

**Comment by clalancette** [comment]
CI:

* Turtlebot [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/98/)

**Review by mikaelarguedas — APPROVED** [review]
lgtm

**Comment by clalancette** [comment]
Thanks, merging.

**Pull request merged**